### PR TITLE
[#149289697] Implement mechanism to allow tests to pass on PRs

### PR DESCRIPTION
**Pull request by DanielBrierton**

### CODE REVIEW

File: build/get_cd_deps.sh [hunk 1]
```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+echo "BUILD_BRANCH: ${BUILD_BRANCH}"
+if [[ $BUILD_BRANCH == "staging" ]]; then
+  echo "Building staging"
+  npm install --save cp-translations@staging @coderdojo/cd-common@staging
+elif [[ $BUILD_BRANCH =~ ^pull\/[0-9]+$ ]]; then
```
Comment (butlerx) [inline build/get_cd_deps.sh]: the `\`  is unneeded

File: build/get_cd_deps.sh [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  echo "Building PR"
+  PR_ID=`echo ${BUILD_BRANCH} | cut -d '/' -f 2`
```
Comment (butlerx) [inline build/get_cd_deps.sh]: we shouls ue `$(echo "${BUILD_BRANCH}" | cut -d '/' -f 2)` rather then ``echo ${BUILD_BRANCH} | cut -d '/' -f 2``

File: build/get_cd_deps.sh [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  PR_JSON=`curl "https://api.github.com/repos/coderdojo/cp-zen-frontend/pulls/${PR_ID}" 2>/dev/null`
+  PR_BRANCH_NAME=`echo $PR_JSON | jq -r '.head.ref'`
```
Comment (butlerx) [inline build/get_cd_deps.sh]: double quote `$PR_JSON` to stop the ourput being globed 
eg `"$PRJSON"`

File: build/get_cd_deps.sh [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  PR_TARGET_BRANCH_NAME=`echo $PR_JSON | jq -r '.base.ref'`
```
Comment (butlerx) [inline build/get_cd_deps.sh]: double quote and `$()`

File: build/get_cd_deps.sh [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  echo "Source branch: ${PR_BRANCH_NAME}"
+  echo "Target branch: ${PR_TARGET_BRANCH_NAME}"
+
+  # cp-translations
+  CP_TRANSLATIONS_BRANCH_STATUS=`curl -I "https://api.github.com/repos/${GITHUB_USERNAME}/cp-translations/branches/${PR_BRANCH_NAME}" 2>/dev/null | head -n 1 | cut -d$' ' -f2`
```
Comment (butlerx) [inline build/get_cd_deps.sh]: `$()`

File: build/get_cd_deps.sh [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+  if [[ $CP_TRANSLATIONS_BRANCH_STATUS == "200" ]]; then
+    echo "Installing cp-translations from ${GITHUB_USERNAME}:${PR_BRANCH_NAME}"
+    npm install git://github.com/${GITHUB_USERNAME}/cp-translations.git#${PR_BRANCH_NAME} --save
```
Comment (butlerx) [inline build/get_cd_deps.sh]: double quote both variables

File: build/get_cd_deps.sh [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+  elif [[ $PR_TARGET_BRANCH_NAME == 'staging' ]]; then
+    echo "Installing cp-translations@staging"
+    npm install --save cp-translations@staging
+  fi
+
+  # cd-common
+  CD_COMMON_BRANCH_STATUS=`curl -I "https://api.github.com/repos/${GITHUB_USERNAME}/cd-common/branches/${PR_BRANCH_NAME}" 2>/dev/null | head -n 1 | cut -d$' ' -f2`
```
Comment (butlerx) [inline build/get_cd_deps.sh]: `$()`

File: build/get_cd_deps.sh [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+  if [[ $CD_COMMON_BRANCH_STATUS == "200" ]]; then
+    echo "Installing cd-common from ${GITHUB_USERNAME}:${PR_BRANCH_NAME}"
+    npm install git://github.com/${GITHUB_USERNAME}/cd-common.git#${PR_BRANCH_NAME} --save
```
Comment (butlerx) [inline build/get_cd_deps.sh]: double quote

File: build/get_cd_deps.sh (hunk 5 above)
Comment (DanielBrierton) [inline build/get_cd_deps.sh]: You mean to use `$()` instead of the backticks? What benefit does that give?

Comment (butlerx) [inline build/get_cd_deps.sh]: yeah. backticks way of being run by bash is worst then `$()`. `$()` will be more reliable and would give the ability to nest commands

File: build/get_cd_deps.sh [hunk 9]
```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
```
Comment (butlerx) [inline build/get_cd_deps.sh]: `#!/usr/bin/env bash` bash isnt always in `/bin`

**Comment by butlerx** [comment]
also might be an idea to add `set -e` to the start of the script

File: build/get_cd_deps.sh [hunk 10]
```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env bash
+set -e
+echo "BUILD_BRANCH: ${BUILD_BRANCH}"
+if [[ $BUILD_BRANCH == "staging" ]]; then
+  echo "Building staging"
+  npm install --save cp-translations@staging @coderdojo/cd-common@staging
+elif [[ "${BUILD_BRANCH}" =~ ^pull/[0-9]+$ ]]; then
+  echo "Building PR"
+  PR_ID=$(echo "${BUILD_BRANCH}" | cut -d '/' -f 2)
+  PR_JSON=$(curl "https://api.github.com/repos/coderdojo/cp-zen-frontend/pulls/${PR_ID}?auth_token=${GITHUB_AUTH_TOKEN}")
```
Comment (butlerx) [inline build/get_cd_deps.sh]: is it not `access_token` ?

Comment (DanielBrierton) [inline build/get_cd_deps.sh]: ........... Yes.... 😅

**Pull request merged**